docs/source/data/subscriptions.mdx

```diff
@@ -0,0 +1,508 @@
+
+Clients can execute this subscription with a GraphQL string like this:
+
+```graphql
+subscription PostFeed {
+  postCreated {
+    author
+    comment
+  }
+}
+```
+
+## Setting a subscription endpoint
+
+Because subscriptions use WebSocket instead of HTTP, Apollo Server uses a _second_ GraphQL endpoint specifically for subscriptions. This endpoint uses the `ws` protocol instead of `http`.
+
+By default, the subscription endpoint's path matches the path of your primary GraphQL endpoint (`/graphql` if not set). You can specify a different path for your subscription endpoint like so:
+
+```js
+const server = new ApolloServer({
+  subscriptions: {
+    path: '/subscriptions'
+  },
+  // ...other options...
+}));
+```
+
+## Resolving a subscription
+
+Resolvers for subscriptions differ from those for queries and mutations. Specifically, subscription resolvers are _objects_ that define a `subscribe` function:
+
+```js:title=index.js
+const resolvers = {
+  Subscription: {
+    postCreated: {
+      // More on pubsub below
+      subscribe: () => pubsub.asyncIterator(['POST_CREATED']),
+    },
+  },
+  // ...other resolvers...
+};
+```
+
+The `subscribe` function must return an object of type `AsyncIterable`, part of Apollo Server's pub/sub API (described below).
+
+### The `PubSub` class
+
+> The `PubSub` class is **not** recommended for production environments, because it's an in-memory event system that only supports a single server instance. After you get subscriptions working in development, we strongly recommend switching it out for a different subclass of the abstract [`PubSubEngine` class](https://github.com/apollographql/graphql-subscriptions/blob/master/src/pubsub-engine.ts). Recommended subclasses are listed in [Production `PubSub` libraries](#production-pubsub-libraries).
+
+Apollo Server uses a **publish-subscribe** (**pub/sub**) model to track events that update active subscriptions. The [`graphql-subscriptions` library](https://github.com/apollographql/graphql-subscriptions) (included in every `apollo-server` package) provides the `PubSub` class to help you get started:
+
+```js
+const { PubSub } = require('apollo-server');
+
+const pubsub = new PubSub();
+```
+
+A `PubSub` instance enables your server code to both `publish` events to a particular label and listen for events associated with a particular label.
+
+### Publishing an event
+
+You publish an event with the `publish` method of a `PubSub` instance:
+
+```js
+pubsub.publish('POST_CREATED', {
+  postCreated: {
+    author: 'Ali Baba',
+    comment: 'Open sesame'
+  }
+});
+```
+
+* The first parameter is the name of the event label you're publishing to, as a string.
+    * _You don't need to register a label name before publishing to it._
+* The second parameter is the payload associated with the event.
+    * _The payload's structure should match the structure that clients expect from the associated subscription's return type._
```

Comment:
It doesn't have to match the structure client's expect though. It's the task of the resolver to map between events and results (what is described as [`MapSourceToResponseEvent` in the spec](http://spec.graphql.org/draft/#sec-Response-Stream)).